polus-multichannel-tiff-plugin/src/main.py

```diff
@@ -0,0 +1,158 @@
+    parser.add_argument('--channelOrder', dest='channelOrder', type=str,
+                        help='Order in which to place the channels (e.g. 1,3,2)', required=True)
+    
+    # Output arguments
+    parser.add_argument('--outDir', dest='outDir', type=str,
+                        help='Output collection', required=True)
+    
+    # Parse the arguments
+    args = parser.parse_args()
+    filePattern = args.filePattern
+    logger.info('filePattern = {}'.format(filePattern))
+    inpDir = args.inpDir
+    if (Path.is_dir(Path(args.inpDir).joinpath('images'))):
+        # switch to images folder if present
+        fpath = str(Path(args.inpDir).joinpath('images').absolute())
+    logger.info('inpDir = {}'.format(Path(inpDir).resolve()))
+    channelOrder = [int(c) for c in str(args.channelOrder).split(',')]
+    logger.info('channelOrder = {}'.format(channelOrder))
+    outDir = args.outDir
+    logger.info('outDir = {}'.format(outDir))
+    
+    # Surround with try/finally for proper error catching
+    try:
+        # Start the javabridge with proper java logging
+        logger.info('Initializing the javabridge...')
+        log_config = Path(__file__).parent.joinpath("log4j.properties")
+        jutil.start_vm(args=["-Dlog4j.configuration=file:{}".format(str(log_config.absolute()))],class_path=bioformats.JARS)
+        
+        # Create a filepattern object to auto sort images
+        fp = filepattern.FilePattern(inpDir,filePattern)
+        
+        # Loop through files in inpDir image collection and process
+        for files in fp.iterate(group_by='c'):
+            
+            # Get the filenames in the channel order
+            paths = []
+            for c in channelOrder:
+                for file in files:
+                    if file['c'] == c:
+                        paths.append(file)
+                        break
+            
+            # make sure that files were found in the current loop
+            if len(paths)==0:
+                continue
+            
+            # Initialize the output file
+            br = BioReader(paths[0]['file'])
+            file_name = filepattern.output_name(filePattern,
+                                                paths,
+                                                {c:paths[0][c] for c in fp.variables if c != 'c'})
+            logger.info('Writing: {}'.format(file_name))
+            bw = BioWriter(str(Path(outDir).joinpath(file_name)),
+                           metadata=br.read_metadata())
+            del br
+            
+            # Modify the metadata to make sure channels are written correctly
+            bw.num_c(len(paths))
+            bw._metadata.image().Pixels.channel_count = bw.num_c()
+            
+            # Process the data in tiles
+            threads = []
+            count = 0
+            total = bw.num_c() * bw.num_z() * \
+                    (bw.num_x()//4096 + 1) * (bw.num_y()//4096 + 1)
```

Comment:
The hardcoded 4096 - can that somehow be extracted from the data rather than hardcoded?

Reply:
Same applies to all occurrences below.